platform/storage/posix/src/mender-storage.c

```diff
@@ -185,6 +186,82 @@ mender_storage_delete_deployment_data(void) {
 #ifdef CONFIG_MENDER_CLIENT_ADD_ON_CONFIGURE
 #ifdef CONFIG_MENDER_CLIENT_CONFIGURE_STORAGE
 
+mender_err_t
+mender_storage_save_update_state(mender_update_state_t state, const char *artifact_type) {
+    assert(artifact_type != NULL);
+    size_t artifact_type_len;
+
+    FILE *f = fopen(MENDER_STORAGE_NVS_UM_STATE, "wb");
+    if (NULL == f) {
+        mender_log_error("Unable to open " MENDER_STORAGE_NVS_UM_STATE " for writing");
+        return MENDER_FAIL;
+    }
+    if (fwrite(state, 1, sizeof(state), f) != sizeof(state)) {
```

Comment:
Incompatible type for `state` and can you use `mender_storage_write_file` and `mender_storage_read_file` for reading and getting?

Reply:
I don't see why when this code is POSIX-specific and can avoid extra memory allocations and such when using POSIX-specific functions.